lib/engine/ability/blocks_crossing_partition.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+require_relative 'base'
+
+module Engine
+  module Ability
+    class BlocksCrossingPartition < Base
```

Comment:
can just call this BlocksPartition, the crossing is inferred because it's a partition